include/stl2/detail/algorithm/generate.hpp

```diff
@@ -20,7 +20,9 @@
 // generate [alg.generate]
 //
 STL2_OPEN_NAMESPACE {
-	template <Callable F, OutputIterator<result_of_t<F&()>> O, Sentinel<O> S>
+	template <class F, OutputIterator<result_of_t<F&()>> O, Sentinel<O> S>
+	requires
+		models::Callable<F&>
```

Comment:
It's too late to require `Callable<F&>`, we've already instantiated `result_of_t<F&()>`. This needs to be:
```c++
template <class F, Iterator O, Sentinel<O> S>
requires
    models::Callable<F&> &&
    models::Writable<O, result_of_t<F&()>>
```